app/navigators/EstablishersCompanyNavigator.scala

```diff
@@ -299,9 +309,17 @@ class EstablishersCompanyNavigator @Inject()(val dataCacheConnector: UserAnswers
       case Some(true) =>
         NavigateTo.dontSave(controllers.register.establishers.company.routes.CompanyVatVariationsController.onPageLoad(mode, index, srn))
       case Some(false) =>
-        NavigateTo.dontSave(controllers.register.establishers.company.routes.HasCompanyPAYEController.onPageLoad(index))
+        NavigateTo.dontSave(controllers.register.establishers.company.routes.HasCompanyPAYEController.onPageLoad(mode, srn, index))
       case None =>
         NavigateTo.dontSave(controllers.routes.SessionExpiredController.onPageLoad())
     }
   }
+
+  private def confirmHasCompanyPAYE(index: Int, mode: Mode, srn: Option[String])(answers: UserAnswers): Option[NavigateTo] =
+    navigateOrSessionExpired(answers, HasCompanyPAYEId(index),
+      if (_: Boolean)
+        establisherCompanyRoutes.CompanyPayeVariationsController.onPageLoad(mode, index, srn)
+      else
+        establisherCompanyRoutes.CheckYourAnswersController.onPageLoad(mode, srn, index)
```

Comment:
New CYA page should be used based on toggle

Reply:
Fixed